### ISSUE TITLE
undo doesn't remember cursor position/selections

### ISSUE THREAD

**Issue by vlovich** [xi-editor/xi-editor#458]
It seems pretty common that if you undo past a selection edit it restores the selection.  XiEditor doesn't seem to do that.

Steps to reproduce:
1. Select a block of text
2. Delete
3. Undo

Expected:
Deleted text re-appears selected

Actual:
Deleted text re-appears but not selected.  In fact cursor position doesn't change ever on undo.

**Comment by raphlinus** [comment]
I think this is the same as #326. It's not simple because undo is done in the CRDT, and selection is not stored in the CRDT, but it could be. One thing to keep in mind is that there will be multiple views to the same buffer, and each view has its own selection.

**Comment by vlovich** [comment]
You're right.  That is a dupe.  As for the multiple views, I'm not sure you need to track each selection/caret move.  Since undo/redo is a modification, on undo/redo you'd change the caret/selection for all views & just make sure the scroll point isn't changed.  At least that's how Xcode, IntelliJ & VS code generally seem to work.

**Comment by cmyr** [comment]
closed as a dupe of #326

**Comment by eyelash** [comment]
I don't think this is a duplicate of #326. This issue is about extending the functionality of the existing undo command whereas #326 is about implementing a new soft undo command.

**Comment by cmyr** [comment]
@eyelash 👍
